lib/manageiq/ui/classic/engine.rb

```diff
@@ -38,14 +38,7 @@ class Engine < ::Rails::Engine
         config.assets.paths << root.join('vendor', 'assets', 'stylesheets').to_s
 
         if Rails.env.production? || Rails.env.test?
-          require 'uglifier'
-          config.assets.js_compressor = Uglifier.new(
-            :compress => {
-              :unused      => false,
-              :keep_fargs  => true,
-              :keep_fnames => true
-            }
-          )
+          config.assets.js_compressor = :uglifier
```

Comment:
Instead of here, lets move these lines to config/environments/production.rb and config/environments/test.rb (which is I think the default location anyway)

Reply:
On that note, do we need to undo any changes to the environments like this one? https://github.com/ManageIQ/manageiq/commit/ec5601fc6a3d61c9e84a0ecbaa906c0144c3dd04

Reply:
> Instead of here, lets move these lines to config/environments/production.rb and config/environments/test.rb (only if that is the normal default location, which is I think it is, but I could be wrong)

Good point... should it be moved?  The dependency is in this repository, not core.  I thought of moving it to core to avoid the conditional here but it feels weird specifying the js_compressor as uglifier in a repo that doesn't have it as a dependency.

Theoretically, if core could support different frontends, they could have different compressors.

I know, right?  I'm in this fantasy world where plugins are silos and things don't leak into core.

Reply:
> On that note, do we need to undo any changes to the environments like this one? https://github.com/ManageIQ/manageiq/commit/ec5601fc6a3d61c9e84a0ecbaa906c0144c3dd04

Honestly, if we're dropping node from production, we can't fallback to the asset pipeline in production.  If it's not compiled and there, it should blow up.

Reply:
ohhh that's an interesting point - I forgot that they were diff repos...technically we have other "UI" settings already in core there, so perhaps core is the right place.

Reply:
> > On that note, do we need to undo any changes to the environments like this one? https://github.com/ManageIQ/manageiq/commit/ec5601fc6a3d61c9e84a0ecbaa906c0144c3dd04
> 
> Honestly, if we're dropping node from production, we can't fallback to the asset pipeline in production. If it's not there, it should blow up.

Sorry - I didn't mean to imply that specifically that commit should be undone, but was using it as an example of changes that were made to config/environments that we might want to review.